base/broadcast.jl

```diff
@@ -316,6 +311,17 @@ end
     end
 end
 @inline broadcast_c(f, ::Type{Any}, a...) = f(a...)
+broadcast_c(f, ::Type{Tuple}, As...) =
+    ntuple(k -> f(_tbcgetargs(As, k)...), _tbcreslength(As))
+broadcast_c{T}(f, ::Type{Tuple}, ::Type{T}, As...) =
+    ntuple(k -> f(T, _tbcgetargs(As, k)...), _tbcreslength(As))
+@inline _tbcgetargs(As, k) = (_broadcast_getindex(first(As), k), _tbcgetargs(tail(As), k)...)
+@inline _tbcgetargs(::Tuple{}, k) = ()
+_tbcreslength(As) = (Base.@_noinline_meta; _tbcmaxlength(_tbclength(first(As)), tail(As)))
+@inline _tbcmaxlength(l, As) = _tbcmaxlength(max(l, _tbclength(first(As))), tail(As))
+@inline _tbcmaxlength(l, ::Tuple{}) = l
+@inline _tbclength(t::Tuple) = length(t)
+@inline _tbclength(s) = 1
```

Comment:
Sorry, I meant the long version. Too long isn't that bad for complex code that appears infrequently and internally - hopefully the greater number of people that can follow the workings of `broadcast`, the better maintained it will be.